### PR TITLE
Allows templating of IAM role policies

### PR DESCRIPTION
**Pull request by danvaida**
In order to avoid duplicated IAM role trust policies and thus keeping code DRY,
we are allowing the templating of such policies.

Upcoming release: `v6.0.0`

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -244,6 +244,8 @@ variable "roles" {
     path                 = string       # Defaults to 'var.role_path' if variable is set to null
     desc                 = string       # Defaults to 'var.role_desc' if variable is set to null
     trust_policy_file    = string       # Path to file of trust/assume policy
```
Comment (cytopia) [inline variables.tf]: Why not simply make the `trust_policy_file` a template file and get rid of the additional `trust_policy_body`. Then you could use optional `trust_policy_vars` with the already existing file variable

Comment (danvaida) [reply variables.tf]: I’d prefer to stick with the separation of concerns, in this case. WDYT?

Comment (cytopia) [reply variables.tf]: I'd stick with `trust_policy_file` and `trust_policy_vars`. This makes it much cleaner.

Downside of having `_file` and `_body` is that from the outside of the module you don't really see what the difference is and that only one of them can be used and also not what the precedence is.

Having just one, omits the user decision and makes it more clear.

Comment (danvaida) [reply variables.tf]: Hey @cytopia, the change has been made, as requested. Please check again.

**Review by ronny-panknin-flaconi — APPROVED** [review]

**Pull request merged**